casper/src/main/scala/coop/rchain/casper/protocol/package.scala

```diff
@@ -0,0 +1,58 @@
+package coop.rchain.casper
+
+import coop.rchain.comm.protocol.routing.Packet
+import coop.rchain.comm.transport
+import scala.util.Try
+
+package object protocol {
+
+  def toCasperMessage(packet: Packet): Option[CasperMessage] =
+    packetToBlockRequest(packet) orElse
+      packetToForkChoiceTipRequest(packet) orElse
+      packetToApprovedBlock(packet) orElse
+      packetToApprovedBlockRequest(packet) orElse
+      packetToBlockMessage(packet) orElse
+      packetToBlockApproval(packet) orElse
+      packetToUnapprovedBlock(packet) orElse
+      packetToNoApprovedBlockAvailable(packet)
+
+  private def packetToBlockMessage(msg: Packet): Option[BlockMessage] =
+    if (msg.typeId == transport.BlockMessage.id)
```

Comment:
The `if (condition) Try(...) else None` logic is repeated many times - it would make sense to extract it to a helper function

Reply:
👍